### PR TITLE
Giftcard 2.0 Datepicker + Resend

### PR DESCRIPTION
**Pull request by philbirt**
* Add datepicker to admin gift card edit
* Add ability to send gift card email from admiin

### CODE REVIEW

**Comment by athal7** [comment]
i think some feature specs could be useful here

**Comment by philbirt** [comment]
Feature spec added

File: spec/features/admin/gift_cards_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      end
+
+      fill_in "virtual_gift_card_recipient_name", with: "Heidi"
+      fill_in "virtual_gift_card_recipient_email", with:"heidi@gmail.com"
+      fill_in "virtual_gift_card_purchaser_name", with: "Neerali"
+      fill_in "virtual_gift_card_gift_message", with: "Sweaters!"
+      fill_in "virtual_gift_card_send_email_at", with: Date.tomorrow
+
+      # Just so the datepicker gets out of poltergeists way.
+      page.execute_script("$('#virtual_gift_card_send_email_at').datepicker('widget').hide();")
```
Comment (athal7) [inline spec/features/admin/gift_cards_spec.rb]: huh?

File: spec/features/admin/gift_cards_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      click_on 'Update'
+      expect(page).to have_content("Gift card updated!")
```
Comment (athal7) [inline spec/features/admin/gift_cards_spec.rb]: do we want to assert against db state in any way?

**Comment by athal7** [comment]
:+1:

File: spec/features/admin/gift_cards_spec.rb (hunk 1 above)
Comment (philbirt) [inline spec/features/admin/gift_cards_spec.rb]: Taken from here: https://github.com/solidusio/solidus/blob/master/backend/spec/features/admin/orders/listing_spec.rb#L102

**Comment by philbirt** [comment]
@athal7 Updated the feature spec with db checks

File: spec/features/admin/gift_cards_spec.rb (hunk 1 above)
Comment (athal7) [inline spec/features/admin/gift_cards_spec.rb]: oic

**Comment by athal7** [comment]
:shipit:

**Pull request merged**